src/main/java/io/seqera/tower/cli/commands/LaunchCmd.java

```diff
@@ -209,6 +229,49 @@ private WorkflowStatus checkWorkflowStatus(String workflowId, Long workspaceId)
         }
     }
 
+    private List<Long> obtainLabelIDs(@Nullable Long workspaceId) throws ApiException {
+
+        if (labels == null || labels.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        // retrieve labels for the workspace and check if we need to create new ones
+        List<LabelDbDto> wspLabels = new ArrayList<>();
+
+        ListLabelsResponse res = api().listLabels(workspaceId, null, null, null, LabelType.SIMPLE, null);
+        if (res.getLabels() != null) {
+            wspLabels.addAll(res.getLabels());
+        }
+
+        Map<String, Long> nameToID = wspLabels
+            .stream()
+            .collect(Collectors.toMap(LabelDbDto::getName, LabelDbDto::getId));
+
+        // get label names not registered in workspace (names are unique per wspID)
+        List<String> newLabels = labels
+            .stream()
+            .filter(labelName -> !nameToID.containsKey(labelName))
+            .collect(Collectors.toList());
+
+        // create the new ones via POST /labels
```

Comment:
Workspace participants with launch role cannot create labels (only use existing ones). For that reason, I don't know to what extent creating labels on the fly should be supported :thinking:.

Reply:
Seems unproductive to me that a Launch user can't create a metadata tag for the pipeline they are invoking.